ledger/accountdb.go

```diff
@@ -1737,6 +1771,32 @@ func (qs *accountsDbQueries) lookupCreator(cidx basics.CreatableIndex, ctype bas
 	return
 }
 
+func (qs *accountsDbQueries) lookupResources(addr basics.Address, aidx basics.CreatableIndex, ctype basics.CreatableType) (data persistedResourcesData, err error) {
+	err = db.Retry(func() error {
+		var buf []byte
+		var rowid sql.NullInt64
+		err := qs.lookupResourcesStmt.QueryRow(addr[:], aidx, ctype).Scan(&rowid, &data.round, &buf)
+		if err == nil {
+			data.aidx = aidx
+			data.rtype = ctype
+			if len(buf) > 0 && rowid.Valid {
+				data.addrid = rowid.Int64
+				return protocol.Decode(buf, &data.data)
+			}
+			// we don't have that account, just return the database round.
+			return nil
+		}
+
+		// this should never happen; it indicates that we don't have a current round in the acctrounds table.
+		if err == sql.ErrNoRows {
+			// Return the zero value of data
+			return fmt.Errorf("unable to query resource data for address %v aidx %v ctype %v : %w", addr, aidx, ctype, err)
```

Comment:
hmm.. error decorating.. I guess it's a good idea to add this, just in case.

Reply:
This is also copy-pasted from lookup() below, it would be a very rare error though.